qiskit/algorithms/amplitude_estimators/estimation_problem.py

```diff
@@ -0,0 +1,180 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Estimation problem class."""
+
+from typing import Optional, List, Callable
+
+from qiskit.circuit import QuantumCircuit
+from qiskit.circuit.library import GroverOperator
+
+
+class EstimationProblem:
+    """The estimation problem is the input to amplitude estimation algorithm.
+
+    This class contains all problem-specific information required to run an amplitude estimation
+    algorithm. That means, it minimally contains the state preparation and the specification
+    of the good state. It can further hold some post processing on the estimation of the amplitude
+    or a custom Grover operator.
+    """
+
+    def __init__(self,
+                 state_preparation: Optional[QuantumCircuit] = None,
```

Comment:
They are none so the object can be incrementally built, maybe you want to set some properties at initialization time and e.g. let a function set the others. Since `state_preparation` and `objective_qubits` are needed by all algos we can make them non-optional though, these are also the one for which we cannot reliably set defaults IMO.

Reply:
Thinking about it, it might be more straightforward to the user to see which arguments are really required. I made `state_preparation` and `objective_qubits` non-optional.